gnome-initial-setup/pages/branding-welcome/gis-branding-welcome-page.c

```diff
@@ -0,0 +1,289 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright (C) 2017 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *
+ * Written by:
+ *     Mario Sanchez Prada <mario@endlessm.com>
+ */
+
+/* Branding Welcome page {{{1 */
+
+#define PAGE_ID "branding-welcome"
+
+#include "config.h"
+#include "branding-welcome-resources.h"
+#include "gis-branding-welcome-page.h"
+
+#include <pango/pango-attributes.h>
+
+struct _GisBrandingWelcomePagePrivate {
+  gchar *title;
+  gchar *description;
+  gchar *logo_path;
+  gchar *skip_pages;
+};
+typedef struct _GisBrandingWelcomePagePrivate GisBrandingWelcomePagePrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (GisBrandingWelcomePage, gis_branding_welcome_page, GIS_TYPE_PAGE);
+
+#define OBJ(type,name) ((type)gtk_builder_get_object(GIS_PAGE(page)->builder,(name)))
+#define WID(name) OBJ(GtkWidget*,name)
+
+/*
+ * CONFIG_FILEPATH points to /etc/eos-branding/gnome-initial-setup/pages.conf, which is a
+ * file containing groups of key-value pairs under a group called "Welcome", defining the
+ * usual parameters needed to render the "branding-welcome" page.
+ *
+ * Among those parameters, the only required one is the "title", meaning that the page will
+ * be forgiving about the rest of them not being there, rendering the title and any other
+ * optional field that it's there, if present.
+ *
+ * Fields:
+ *   - title (required): string containing a short string to show as title.
+ *   - description (optional): string containing long text, likely to be wrapped.
+ *   - logo (optional): absolute path to the file with a logo for the brand.
+ *   - skip (optional): comma-separated list of PAGE IDs the branded g-i-s should skip.
+ *
+ * For example, this is how this file would look on a branded image defining a title,
+ * a description, a logo and skipping the "language" and "keyboard" pages.
+ *
+ *   [Welcome]
+ *   title=A title to be shown at the top
+ *   description=A long description that will be shown at the bottom of this page, right below the branded logo (if any), explaining what the branded edition is about.
+ *   logo=/path/to/the/image/with/the/logo.png
+ *   skip=language,keyboard
+ */
+#define CONFIG_FILEPATH CONFIGDIR "/eos-branding/gnome-initial-setup/pages.conf"
```

Comment:
The reason I am not a huge fan of it is because it makes it impossible to upstream it; I think that if we used different paths for the customization point (or if we made the path for the customization point itself also configurable e.g. through a build option) we could make a case for these to also go upstream.

However if you're short on time, I'd be OK accepting this as-is.